### PR TITLE
Handle `_table` and `_column` suffixes on migration creation.

### PR DESCRIPTION
**Pull request by rob006**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues | - |

This supports more explicit migration names, like:
- `create_user_table`,
- `add_name_column_to_user_table`,
- `create_junction_table_for_post_and_tag_tables`.

### CODE REVIEW

**Comment by pana1990** [comment]
this pr is a bugfix, it fix https://github.com/yiisoft/yii2/issues/10876

It neccesary update docs and it changes to drop tables and drop columns syntax.

i suggest the following syntax :

- drop_xxx_from_yyy to dropcolumn_xxx_from_yyy or dropColumn_xxx_yyy

- drop_xxx

 to droptable_xxx or dropTable_xxx

What do you about it?

i can help  you with docs and check code

**Comment by rob006** [comment]
This PR still support old short syntax, so if #10876 is a bug, this PR will not fix it. My intention was only to remove annoying suffixes from generated migrations, for example `./yii migrate/create create_user_table` generate:

```php
<?php

use yii\db\Migration;

/**
 * Handles the creation for table `user_table`.
 */
class m160623_141619_create_user_table extends Migration
{
    /**
     * @inheritdoc
     */
    public function up()
    {
        $this->createTable('user_table', [
            'id' => $this->primaryKey(),
        ]);
    }

    /**
     * @inheritdoc
     */
    public function down()
    {
        $this->dropTable('user_table');
    }
}
```

Your solution does not really solve this issue. You can still have column like `name_from_api` and migration like `dropcolumn_name_from_api_from_user`. IMO `_column` and `_table` suffixes should be mandatory, but @yiisoft/core-developers should decide.

**Comment by pana1990** [comment]
ups yeah, you are right

File: tests/framework/console/controllers/MigrateControllerTestTrait.php [hunk 1]
```diff
@@ -194,93 +194,121 @@ public function testGenerateDefaultMigration()
...
     {
-        $migrationName = 'create_junction_post_and_tag';
-        $this->assertCommandCreatedFile('junction_test', $migrationName);
+        $migrationNames = [
+            'create_junction_post_and_tag',
+            'create_junction_table_for_post_and_tag_tables',
+            'create_junction_table_for_post_and_tag_table',
+            'create_junction_post_and_tag_tables',
+        ];
+        foreach($migrationNames as $migrationName) {
```
Comment (samdark) [inline tests/framework/console/controllers/MigrateControllerTestTrait.php]: `foreach (`

File: tests/framework/console/controllers/MigrateControllerTestTrait.php [hunk 2]
```diff
@@ -194,93 +194,121 @@ public function testGenerateDefaultMigration()
...
-            'fields' => 'title:string(10):notNull,body:text:notNull,
-                price:money(11,2):notNull,
-                created_at:dateTime'
-        ]);
+        $migrationNames = [
+            'drop_columns_from_test',
+            'drop_columns_from_test_table',
+            'drop_columns_column_from_test_table',
+        ];
+        foreach($migrationNames as $migrationName) {
```
Comment (samdark) [inline tests/framework/console/controllers/MigrateControllerTestTrait.php]: `foreach (`

File: tests/framework/console/controllers/MigrateControllerTestTrait.php [hunk 3]
```diff
@@ -194,93 +194,121 @@ public function testGenerateDefaultMigration()
...
-        $this->assertCommandCreatedFile('drop_test', $migrationName);
-
-        $this->assertCommandCreatedFile('drop_fields', $migrationName, [
-            'fields' => 'body:text:notNull,price:money(11,2)'
-        ]);
+        $migrationNames = [
+            'drop_test',
+            'drop_test_table',
+        ];
+        foreach($migrationNames as $migrationName) {
```
Comment (samdark) [inline tests/framework/console/controllers/MigrateControllerTestTrait.php]: `foreach (`

File: tests/framework/console/controllers/MigrateControllerTestTrait.php [hunk 4]
```diff
@@ -194,93 +194,121 @@ public function testGenerateDefaultMigration()
...
-            'fields' => 'user_id:integer:foreignKey,
-                product_id:foreignKey:integer:unsigned:notNull,
-                order_id:integer:foreignKey(user_order):notNull,
-                created_at:dateTime:notNull',
-        ]);
+        $migrationNames = [
+            'create_test',
+            'create_test_table',
+        ];
+        foreach($migrationNames as $migrationName) {
```
Comment (samdark) [inline tests/framework/console/controllers/MigrateControllerTestTrait.php]: `foreach (`

**Comment by samdark** [comment]
Need a line for changelog. Looks OK to me otherwise.

**Comment by rob006** [comment]
@samdark What about #10876? We could close it if we remove support for short syntax without `_table` and `_column` suffixes. Or just close #10876 as it is edge case which is not supported and user should fix migration himself (it's just a template and does not need to be 100% bulletproof).

**Comment by samdark** [comment]
I think we should remove short syntax.

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**